### PR TITLE
Framework: Remove i18n mixin injections from tests

### PR DESCRIPTION
**Pull request by ockham**
Remove i18n mixin injections from tests (and some obsolete `matches-selector` stubs I found while working on this). This PR really only touches test files -- the corresponding units have been using `localize` already, only the tests hadn't been updated accordingly yet.

Rationale: We're trying to get rid of mixin auto-injection for the React 16 upgrade, since this relies on methods (`react/lib`) that aren't part of the public interface.

To test: Verify that unit tests pass.

[Follow-up PR](https://github.com/Automattic/wp-calypso/pull/18472) that tackles `client/auth/test/login.jsx` which also requires changes to the tested unit.

### CODE REVIEW

File: client/my-sites/domains/domain-management/edit/test/mapped-domain.js [hunk 1]
```diff
@@ -3,15 +3,19 @@ jest.mock( 'lib/analytics', () => {} );
...
  * External dependencies
  */
+import React from 'react';
+import TestUtils from 'react-addons-test-utils';
 import assert from 'assert';
 import sinon from 'sinon';
 import { identity } from 'lodash';
 
+/**
+ * External dependencies
```
Comment (samouri) [inline client/my-sites/domains/domain-management/edit/test/mapped-domain.js]: Lies!

**Comment by samouri** [comment]
I don't really know the old paradigm of mixin injection for tests. You are deleting code though and tests still pass...so seems good to me!

**Comment by ockham** [comment]
> I don't really know the old paradigm of mixin injection for tests. You are deleting setup code and tests still pass...so seems good to me!

@samauri I think the reason is simply that

> the corresponding units have been using `localize` already, only the tests hadn't been updated accordingly yet.

File: client/my-sites/domains/domain-management/edit/test/mapped-domain.js (hunk 1 above)
Comment (ockham) [reply client/my-sites/domains/domain-management/edit/test/mapped-domain.js]: Good spot. 2c2f670

**Pull request merged**